deployment/terraform/engine.go

```diff
@@ -85,3 +87,24 @@ func (t *Terraform) runCommand(dst io.Writer, args ...string) error {
 	}
 	return nil
 }
+
+func checkTerraformVersion() error {
+	cmd := exec.Command("terraform", "version")
+	out, err := cmd.Output()
+	if err != nil {
+		return fmt.Errorf("could not run terraform version command: %w", err)
+	}
+	re := regexp.MustCompile(`v\d+.?\d+`)
+	if !re.Match(out) {
+		return fmt.Errorf("could not parse terraform command output: %s", out)
+	}
+	match := re.FindAll(out, 1)[0]
+	version, err := strconv.ParseFloat(string(match[1:]), 64)
+	if err != nil {
+		return fmt.Errorf("could not parse terraform command output: %w", err)
+	}
+	if version < 0.12 {
+		return fmt.Errorf("at least terraform v0.12 is required, you have %s", match)
+	}
```

Comment:
Wondering if we really need to go to the trouble of converting to `float`. I think a check for `v0.12` prefix would be enough. We can't guarantee we support a possible 0.13 either since it might break things.

Reply:
I was not so sure about this, will change it 👍